### PR TITLE
[SYCL][FPGA] Fix the way we handle duplicate vs conflicting values with loop attributes

### PR DESCRIPTION
**Pull request by smanna12**
This patch improves diagnostic supports by resolving bugs the way we handle duplicate vs conflicting values with the following SYCL FPGA loop attributes:  

      [[intel::max_reinvocation_delay()]]
      [[intel::initiation_interval()]]
      [[intel::max_concurrency()]]
      [[intel::speculated_iterations()]]
      [[intel::max_interleaving()]]

The patch addresses issues in the test case below, which previously missed diagnostics due to a discontinuation in the while loop while checking for duplicate versus conflicting attribute values in the routine CheckForDuplicateAttrs().

Example with `speculated_iterations' attribute:

       Before the fix:

        [[intel::speculated_iterations(1)]] // expected-note {{previous attribute is here}}
        [[intel::speculated_iterations(1)]] // OK
        [[intel::speculated_iterations(2)]] // expected-error {{conflicting loop attribute 'speculated_iterations'}}
        [[intel::speculated_iterations(4)]] // OK
        for (int i = 0; i != 10; ++i) { a[i] = 0; }

        After the fix:

        [[intel::speculated_iterations(1)]] // expected-note 2 {{previous attribute is here}}
        [[intel::speculated_iterations(1)]] // OK
        [[intel::speculated_iterations(2)]] // expected-error {{conflicting loop attribute 'speculated_iterations'}}
        [[intel::speculated_iterations(4)]] // expected-error {{conflicting loop attribute 'speculated_iterations'}}
        for (int i = 0; i != 10; ++i) { a[i] = 0; }

 Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: clang/lib/Sema/SemaStmtAttr.cpp [hunk 1]
```diff
@@ -1025,9 +1025,9 @@ static void CheckForDuplicateAttrs(Sema &S, ArrayRef<const Attr *> Attrs) {
       S.Diag((*LastFoundItr)->getLocation(), diag::err_loop_attr_conflict)
           << *FirstItr;
       S.Diag((*FirstItr)->getLocation(), diag::note_previous_attribute);
-      return;
     }
   }
+  return;
```
Comment (premanandrao) [inline clang/lib/Sema/SemaStmtAttr.cpp:1030]:
```suggestion
```

Comment (smanna12) [reply clang/lib/Sema/SemaStmtAttr.cpp]: Thanks for the reviews @premanandrao. Done

**Review by premanandrao — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you @premanandrao for reviews!

**Comment by smanna12** [comment]
@intel/llvm-gatekeepers, this PR is ready to merge. Thank you

**Pull request merged**